oio/event/filters/meta2_index.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (C) 2018 OpenIO SAS, as part of OpenIO SDS
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from oio.common.exceptions import ClientException, OioTimeout, VolumeException
+from oio.common.logger import get_logger
+from oio.account.client import AccountClient
+from oio.rdir.client import RdirClient
+from oio.event.evob import Event, EventError
+from oio.event.consumer import EventTypes
+from oio.event.filters.base import Filter
+
+# We still don't have a working implementation of META2 assignment to RDIR.
+# So we keep this in passthrough mode for the moment.
+
+PASSTHROUGH = False
+
+
+class Meta2IndexFilter(Filter):
+
+    def __init__(self, app, conf, **kwargs):
+        self.logger = get_logger(conf)
+        super(Meta2IndexFilter, self).__init__(app, conf,
+                                               logger=self.logger, **kwargs)
+        self.account = AccountClient(conf, logger=self.logger)
+        self.rdir = RdirClient(conf, logger=self.logger)
+
+    def process(self, env, cb):
+        event = Event(env)
+
+        mtime = event.when / 1000000.0
+        url = event.env.get('url')
+        content_url = "/".join([
+            url.get('ns'),
+            url.get('account'),
+            url.get('user')])
```

Comment:
`encode_fullpath` is not usable in this case because we're trying to encode a string in the form of `NS/account/container` while the function is used to encode strings in the form of `account/container/object/version/content_id`.